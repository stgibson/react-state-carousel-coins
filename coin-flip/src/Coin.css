.Coin {
  margin: 10px;
}