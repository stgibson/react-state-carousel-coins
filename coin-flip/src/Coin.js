import React from "react";
import "./Coin.css";

const Coin = (props) => {
  return (
    <div className="Coin">{ props.side === "h" ? "Heads" : "Tails" }</div>
  );
};

export default Coin;