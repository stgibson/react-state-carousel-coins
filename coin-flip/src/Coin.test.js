import React from "react";
import { render } from "@testing-library/react";
import Coin from "./Coin";

it("loads as heads without crashing", function() {
  render(<Coin side="h" />);
});

it("loads as tails without crashing", function() {
  render(<Coin side="t" />);
});

it("matches snapshot as heads", function() {
  const { asFragment } = render(<Coin side="h" />);
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot as tails", function() {
  const { asFragment } = render(<Coin side="h" />);
  expect(asFragment()).toMatchSnapshot();
});