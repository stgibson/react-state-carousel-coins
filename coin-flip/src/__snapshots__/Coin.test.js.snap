// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot as heads 1`] = `
<DocumentFragment>
  <div
    class="Coin"
  >
    Heads
  </div>
</DocumentFragment>
`;

exports[`matches snapshot as tails 1`] = `
<DocumentFragment>
  <div
    class="Coin"
  >
    Heads
  </div>
</DocumentFragment>
`;
